### DISCUSSION TITLE
Debug mono on Linux

### DISCUSSION THREAD

**Discussion by pedrobsaila** [dotnet/runtime#67620] [Q&A] [answered]
I need to debug mono https://github.com/dotnet/runtime/tree/main/src/mono. Does someone know how to launch a console dotnet program with a locally built mono on Linux (Debian 10) and debug it ?

I will be grateful for any commands/hints/docs you can provide.

**Comment by huoyaoyuan** [comment, marked as answer]
In fact, corerun for coreclr can actually load Mono too. You can follow the coreclr debugging instructions, and set working directory or CORE_ROOT environment variable to build output of Mono. The Mono runtime entry point is also named as `coreclr.dll` (`libcoreclr.so`, but I'm not very sure for Linux) for this.

**Reply by pedrobsaila** [reply]
Much thanks @huoyaoyuan, that works nicely
